src/App.js

```diff
@@ -1,26 +1,92 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import { Home, Quote, Searchbar } from './components';
+import './App.scss';
+import movies from './movies/index';
+import Fuse from 'fuse.js';
 
-function App() {
-  return (
+const buildContext = (movie, index) => {
+    const start = index - 10 >= 0 ? index - 10 : 0;
+    const end = index + 10 >= movie.subs.length ? movie.subs.length : index + 10;
+
+    return {
+      prev: movie.subs.slice(start, index),
+      post: movie.subs.slice(index + 1, end + 1),
+    };
+};
+
+export default function App() {
+    let quoteFromLocation;
+    try {
+      let queryParams = new URLSearchParams(document.location.search);
+      let indexFromUrl = parseInt(queryParams.get("quoteIndex"), 10);
+      let movieFromUrl = queryParams.get("movie");
+      if (movieFromUrl && indexFromUrl) {
+        let movie = movies.find(movie => movie.id === movieFromUrl);
+        if (movie) {
+          quoteFromLocation = {
+            movie: movie,
+            sub: movie.subs[indexFromUrl],
+            index: indexFromUrl,
+          };
+        }
+      }
+    } catch (e) {}
+
+    let flatSubs = movies.map(x => x.subs.map((y,i) => ({"id": x.id, 'sub':y.sub.join(" "), 'i':i}))).flatMap(x => x);
```

Comment:
this is flatmap in disguise, you are mapping and then flatten it using an identity (`.flatMap(x => x)` === `.flat()`

```suggestion
    const flatSubs = movies.flatMap(x => x.subs.map((y,i) => ({"id": x.id, 'sub':y.sub.join(" "), 'i':i})))
```